sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -223,11 +223,57 @@ private[sql] object JDBCRDD extends Logging {
         } else {
           null
         }
+      case ArithmeticOPEqualTo(operation, value) =>
+        getArithmeticString(operation).get + s" =  ${compileValue(value)}"
```

Comment:
When `getArithmeticString` returns `None`, you will get `java.util.NoSuchElementException` here.